examples/validate-json-with-zod/astro.config.mjs

```diff
@@ -0,0 +1,20 @@
+import { defineConfig } from 'astro/config';
+import { z } from 'zod';
+
+// https://astro.build/config
+export default defineConfig({
+	jsonDataFiles: [
+		{
+			path: 'src/data.json',
+			schema: z.object({
+				name: z.string(),
+				images: z.array(
+					z.object({
+						imageSrc: z.string(), // TODO: use an optimized image type
+						imageAlt: z.string(),
+					})
+				),
+			}),
+		},
+	],
```

Comment:
Ah, this direction surprised me! 

We already have a pattern for type-safe/validated [data collections](https://docs.astro.build/en/guides/content-collections/#defining-a-collection-schema) using the `astro:content` module.

Reply:
Don't you think that having a data collection with 1 JSON file for a name, description, and heroImage of a website would be confusing? The alternative is that I take the same code, but put the vite plugin code that I've added (in astro-postprocess) in my own astro project, which seems like something that astro would/should handle 🤔.  If that option does end up being taken (IE no singular JSON file validation with zod), then I will still need to know how images get processed, as I can't find anything that exports `z` within `/packages/astro/content/index.ts`, so don't know where to start looking when faced with this line of code:
```ts
import { z, defineCollection } from 'astro:content';
```

Reply:
I'm not really suggesting that you should use a full collection to handle a single file! I'm suggesting that if we want to implement this feature, the API should align with the existing ones. In this case, having configuration for validating a single file in the `astro.config.mjs` file but configuration for collection validation in the `src/content/config.ts` seems pretty confusing.

We can bikeshed on the exact naming, but it might be able to support a single argument version of `getEntry()`? So something like `getEntry("site-info")` would correspond to "src/content/site-info.json".

Reply:
I'd also say that we _probably_ don't need to modify the existing content collections logic that much. If we can just wire up the existing collection resolution logic, we won't need to implement a whole new Vite plugin for loading these files.